packages/auth/src/helpers.ts

```diff
@@ -190,20 +190,45 @@ export async function getValidTokenFromProvider(
 	throw lastTokenError ?? new Error('Could not retrieve a valid token');
 }
 
+const scopeEquivalencies = new Map([
+	['channel_commercial', 'channel:edit:commercial'],
+	['channel_editor', 'channel:manage:broadcast'],
+	['channel_read', 'channel:read:stream_key'],
+	['channel_subscriptions', 'channel:read:subscriptions'],
+	['user_blocks_read', 'user:read:blocked_users'],
+	['user_blocks_edit', 'user:manage:blocked_users'],
+	['user_follows_edit', 'user:edit:follows'],
+	['user_read', 'user:read:email'],
+	['user_subscriptions', 'user:read:subscriptions'],
+	['user:edit:broadcast', 'channel:manage:broadcast'],
+]);
+
 /**
  * Compares scopes for a non-upgradable `AuthProvider` instance.
  *
  * @param scopesToCompare The scopes to compare against.
  * @param requestedScopes The scopes you requested.
  */
 export function compareScopes(scopesToCompare: string[], requestedScopes?: string[]): void {
```

Comment:
Note: in the future, if there is ever an endpoint that accepts `channel:manage:broadcast` but not `user:edit:broadcast`, `strict: boolean = false` could be added to method definitions to skip checking equivalencies when true

Reply:
Not a pressing issue right now. From past experience with the Twitch API, if they add something new with at least half a year difference to the last thing in a generally similar direction, they'll just add a new scope.